### ISSUE TITLE
Edits applied to one cell vs applied to all similar cells : different behavior on record creation

### ISSUE THREAD

**Issue by msaby** [OpenRefine/OpenRefine#1991]
**Describe the bug**

If you edit an individual cell of the 1st column with "Edit this cell", and replace its value with "", it does not create a new record.

If you do the same operation but apply the changes to all similar cells, new records are created.

**To Reproduce**

original data :

![image](https://user-images.githubusercontent.com/932804/54877794-e687eb80-4e23-11e9-9ce7-0fdc173b751a.png)

edit the last cell : replace "z" with "", and apply the changes only this cell

![image](https://user-images.githubusercontent.com/932804/54877809-0fa87c00-4e24-11e9-8e59-9b0c52181abd.png)

result : 
![image](https://user-images.githubusercontent.com/932804/54877811-1931e400-4e24-11e9-97c1-54fca83da9bf.png)

Same operation, but apply the changes to all similar cells : 

![image](https://user-images.githubusercontent.com/932804/54877814-2e0e7780-4e24-11e9-98c5-b104df7ceb01.png)

**Current Results**
What results occured or were shown.

**Expected behavior**

The behavior should be the same... If we accept the current behavior of creating new records when a cell contains "" (and not only null)

**Screenshots**

**Desktop (please complete the following information):**
 - OS: [e.g. iOS, Windows 10, Linux, Ubuntu 18.04]
 - Browser Version: [e.g. Chrome 19, Firefox 61, Safari, NOTE: OpenRefine does not support IE but works OK in most cases]
 - JRE or JDK Version:[output of "java -version" e.g. JRE 1.8.0_181]

**OpenRefine (please complete the following information):**
 - Version : Openrefine 3.1

**Datasets**
If you are allowed and are OK with making your data public, it would be awesome if you can include or attach the data causing the issue or a URL pointing to where the data is.
If you are concerned about keeping your data private, ping us on our [mailing list](https://groups.google.com/forum/#!forum/openrefine)

**Additional context**

**Comment by nanobrad** [comment]
I notice that the first example (edit single cell) will *not* recognize that it should be a record--even refreshing the page or faceting will not cause it to reload and show as a record.

Isolating the "blank" cell (probably an empty string) and applying Column->Edit cells->Common transforms->To null will cause the the row to be incorporated into the record.

Rolling back the undo to the "Create project" will not break up the record. It appears you have to restart openrefine.exe to get it to reevaluate the record situation.

Example:
![image](https://user-images.githubusercontent.com/597719/55977164-bb8afd80-5c53-11e9-91d9-78ccda0bf4e9.png)

Using Apply to All Identical Cells works as expected. The undo works in this case as well.

In summary, there is something wonky with the record caching.
